src/bootstrap/bootstrap.py

```diff
@@ -393,6 +393,7 @@ def download_toolchain(self, stage0=True, rustc_channel=None):
         Each downloaded tarball is extracted, after that, the script
         will move all the content to the right place.
         """
+        print("Downloading toolchain (if necessary)")
```

Comment:
I would only print this if bootstrap actually decides to download the toolchain.

Reply:
I was following the lead of submodule updating, which always prints something...

Reply:
Personally that annoys me too, I would rather it only print that if it updated a submodule. Not sure how hard that is to change.

Reply:
It's hard to change. I've just removed it.

Reply:
Hmm, I think the check just needs to be `if not stage0: print("Downloading toolchain")`

Reply:
This doesn't work, at least not how I expected it would. I tried adding it, then did `./x.py clean ; ./x.py build` and it didn't print "Downloading toolchain". Then I tried `rm -rf build/ ; ./x.py build" and again it didn't print "Downloading toolchain".

I think the PR is a sufficient improvement as is, and is ready to be merged.